### PR TITLE
Add importPFB tests [SATURN-987]

### PR DESCRIPTION
**Pull request by breilly2**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala:1088]: my *opinion* is that if Arrow throws a 400, orch should return a 400. In other words, if Arrow tells us the user made a bad request, we should  tell the user they made a bad request instead of telling the user that something failed on the server side. I won't block the PR  on this  - it's your call.

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala:1152]: it's less clear to me what to return here. If Rawls returns a 400, I'm still tempted to return a 400 to the user, though I can certainly see why it might actually be a 500.

**Review by davidangb — COMMENTED** [review]
minor comments inline.

Please add a test for a `PfbImportRequest` that contains a space and/or other extended characters, since that was a minor sticking point during development.

Another corner case to test is what happens if the user's PFB is extremely large and Arrow returns a huge amount of JSON. I suspect this case will be handled in various error trapping, but maybe worth a test?

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala
Comment (breilly2) [reply src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala:1088]: There are a total of 4 requests in this whole process so it gets confusing.

* client -> Orch: If Orch can't understand the request, that is definitely the client's responsibility.
* Orch -> Arrow: If Orch does a poor job converting query parameters into the JSON that Arrow is expecting, this is a bad request that is Orch's fault, not the client's.
* Orch -> Rawls: Orch is making the call, but the guts of the request comes from Arrow. A bad request is perhaps a little bit on Orch but mostly on Arrow. Again, it's not the client's responsibility.
* Arrow -> Avro file: This is the most questionable. Arrow is the one actually making the request, but it's requesting exactly what was passed to Arrow from the client. That could be the client's responsibility. Should Arrow return 400 or 500 for this? If 400, what should Orch do with it?

There are definitely two distinct legitimate reasons why Arrow could return 400. As long as the body contains enough detail to disambiguate between the cases, we can probably choose either. So how do we choose?

I guess 400 would be something the client could potentially do something to fix themselves. Therefore, the answers to the questions are that Arrow and Orch should both lift a 400 from Arrow -> Avro file. This will potentially also lift Orch -> Arrow 400s, but the error content can indicate that it's actually not the client's fault. I will make this change.

**Comment by davidangb** [comment]
or better yet - add a unit test for a user posting the empty string as a url to this API:
```
{"url":""}
```

**Comment by breilly2** [comment]
The space encoding thing was sketchy to begin with. It was a response to a specific case of bad input because, at the time, we were able to respond with that fix quicker than the fix could be applied on the other end. Fully encoding the input breaks things when the input is already properly encoded (which is why the space encoding was a sketchy text substitution).

This time around, I haven't seen evidence of encoding problems and will push back on clients to send proper input if it comes up.

**Comment by breilly2** [comment]
Huge input may be a problem at some point, but I think that would need to be more of an integration test and/or stress tests on Arrow and Rawls. The only problem I can think of local to Orch would be consuming huge amounts of memory because of having to keep the JSON in memory while passing it from Arrow to Rawls. That would definitely require an integration test.

(I'm not sure if using the Arrow response `HttpEntity` as the input `HttpEntity` to Rawls allows spray to stream it through, but it can't hurt to avoid deserializing the JSON into objects just to serialize them back out again.)

**Comment by breilly2** [comment]
Also, what the heck coveralls!? I _added_ tests! *:sigh:*

**Review by davidangb — APPROVED** [review]
:+1: lgtm!

**Comment by breilly2** [comment]
One test failed with a `bucket did not become readable` error, so... jenkins retest!?

**Pull request merged**
